exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetComparisonPredicate.java

```diff
@@ -83,23 +84,27 @@ private ParquetComparisonPredicate(
    * where Column1 and Column2 are from same parquet table.
    */
   @Override
-  public boolean canDrop(RangeExprEvaluator<C> evaluator) {
+  public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
     Statistics<C> leftStat = left.accept(evaluator, null);
     if (isNullOrEmpty(leftStat)) {
-      return false;
+      return RowsMatch.SOME;
     }
-
     Statistics<C> rightStat = right.accept(evaluator, null);
     if (isNullOrEmpty(rightStat)) {
-      return false;
+      return RowsMatch.SOME;
     }
-
-    // if either side is ALL null, = is evaluated to UNKNOWN -> canDrop
-    if (isAllNulls(leftStat, evaluator.getRowCount()) || isAllNulls(rightStat, evaluator.getRowCount())) {
-      return true;
+    if (isAllNulls(leftStat, evaluator.getRowCount()) || isAllNulls(rightStat, evaluator.getRowCount())
+      || !leftStat.hasNonNullValue() || !rightStat.hasNonNullValue()) {
```

Comment:
I am not sure that `NONE` is the right result if min and max are not available. Should it be `SOME`?

Reply:
Agree. For my parquet files it makes of course no difference, since they are generated straight from scala/spark code. But in case the min/max are not defined but the number of null is, then we may want to apply the filter, for security.
Change added.